modules/oe_translation_cdt/tests/src/Kernel/CdtApiAuthenticatorTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+use Drupal\Component\Datetime\Time;
+use Drupal\Core\State\StateInterface;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\oe_translation_cdt\Api\CdtApiAuthenticator;
+use Drupal\oe_translation_cdt\Exception\CdtConnectionException;
+use OpenEuropa\CdtClient\Contract\ApiClientInterface;
+use OpenEuropa\CdtClient\Model\Response\Token;
+use Prophecy\Prophecy\ObjectProphecy;
+
+/**
+ * Tests the CDT API client wrapper.
+ *
+ * @group batch1
+ */
+class CdtApiAuthenticatorTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_translation_cdt',
+  ];
+
+  /**
+   * The API client prophecy.
+   */
+  private ApiClientInterface|ObjectProphecy $apiProphecy;
```

Comment:
this is a bit confusing for me and my editor, this returns Prophecy\Prophecy\ObjectProphecy

Reply:
I took this approach [from core](https://git.drupalcode.org/project/drupal/-/blob/10.3.x/core/modules/field/tests/src/Unit/Plugin/migrate/process/ProcessFieldTest.php?ref_type=heads#L34). Unfortunately, PHPStorm doesn't like Prophecy by default. It requires a [plugin](https://plugins.jetbrains.com/plugin/9674-phpunit-enhancement) to properly understand it.